### PR TITLE
OpInfo: Added clamp and trunc tests with aliases

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- Added clamp, trunc tests with aliases
- Added tests for aliases for asin(h), acos(h), etc
- fixed 'fix' alias implementation
- fixed annotations in test_jit_alias_remapping
- updated native_functions.yaml aliases guidelines

Blocked by #50368

cc @mruberry

### CODE REVIEW

**Comment by mruberry** [comment]
This looks good. It should also cleanup the clamp metadata here:

https://github.com/pytorch/pytorch/blob/0b5303e833ed7e46e5aad4e1770f9bf4ef7d6c6b/test/test_torch.py#L6617

and the trunc metadata here:

https://github.com/pytorch/pytorch/blob/0b5303e833ed7e46e5aad4e1770f9bf4ef7d6c6b/test/test_torch.py#L6855

and here:

https://github.com/pytorch/pytorch/blob/0b5303e833ed7e46e5aad4e1770f9bf4ef7d6c6b/torch/testing/_internal/common_methods_invocations.py#L2133

**Comment by mruberry** [comment]
Most test failures are in base and should be fixed if this rebases. The clamp failures like this are real, though:

```
test_batch_vs_slicing_clamp_cpu_int16 - TestUnaryUfuncsCPU
test_unary_ufuncs.py

Traceback (most recent call last):
  File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_device_type.py", line 295, in instantiated_test
    raise rte
  File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_device_type.py", line 290, in instantiated_test
    result = test_fn(self, *args)
  File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_device_type.py", line 268, in test_wrapper
    return test(*args, **kwargs)
  File "test_unary_ufuncs.py", line 400, in test_batch_vs_slicing
    actual = op(input)
  File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_methods_invocations.py", line 217, in __call__
    return self.op(*args, **kwargs)
RuntimeError: At least one of 'min' or 'max' must not be None
```

Should be an easy fix.

**Comment by mruberry** [comment]
And these can be removed, too:

https://github.com/pytorch/pytorch/blob/4a2aa0f5f1572d8a9caca9496669f8f52cfa1522/test/test_unary_ufuncs.py#L1685

https://github.com/pytorch/pytorch/blob/4a2aa0f5f1572d8a9caca9496669f8f52cfa1522/test/test_unary_ufuncs.py#L1698

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -2510,13 +2546,6 @@ def method_tests():
         ('view_as_real', (S, S, S), NO_ARGS, 'complex'),
         ('view_as_complex', (S, S, 2), NO_ARGS),
         ('complex', (S, S, S), ((S, S, S),), ''),
-        ('clamp', (S, S, S), (0, 1), '', (True,)),
-        ('clamp', (S, S, S), (None, 0.5), 'min', (True,)),
-        ('clamp', (S, S, S), (0.5, None), 'max', (True,)),
-        ('clamp', (), (0, 1), 'scalar', (True,)),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:2516]: Should we keep this empty tensor case for clamp's sample inputs?

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -1256,6 +1267,7 @@ def sample_inputs_masked_select(op_info, device, dtype, requires_grad):
                    )),
     # NOTE: the derivative for inplace acosh is not implemented
     UnaryUfuncInfo('acosh',
+                   aliases=('arccosh', ),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1270]: These aliases can have their AliasInfos removed from:

https://github.com/pytorch/pytorch/blob/f7313b310505df5593072d17b7f60f516bcf029a/test/test_op_aliases.py#L36

File: test/test_ops.py [hunk 3]
```diff
@@ -370,10 +370,13 @@ def test_jit_alias_remapping(self, device, dtype, op):
         sample = samples[0]
 
         # Prepare data for test scripting
-        args = [f"t{i}" for i in range(len(sample.input))] + \
-               [f"s{i}" for i in range(len(sample.args))]
-        args_annot_kw = args + [f"{k}: {type(v).__name__} = {v}" for k, v in sample.kwargs.items()]
-        args_kw = args + [f"{k}={v}" for k, v in sample.kwargs.items()]
+        args = [f"t{i}" for i in range(len(sample.input))]
```
Comment (mruberry) [inline test/test_ops.py:373]: Elaborate in this comment on how the args are being prepared.

**Comment by mruberry** [comment]
Looking good, @vfdev-5, I made just a few comments inline!

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (vfdev-5) [reply torch/testing/_internal/common_methods_invocations.py:1270]: I was thinking that we will remove all AliasInfos from test_op_aliases.py once everything is ported ?

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (vfdev-5) [reply torch/testing/_internal/common_methods_invocations.py:2516]: OK, I added creation of empty tensor and similar inputs as 
```
('clamp', (S, S, S), (None, 0.5), 'min', (True,)),
('clamp', (S, S, S), (0.5, None), 'max', (True,)),
```

**Review by mruberry — APPROVED** [review]
Great attention to detail, removal of legacy tests, and an awesome fix for fix! Thanks @vfdev-5!

**Pull request closed without merging**